dhall-docs/src/Dhall/Docs/Core.hs

```diff
@@ -85,6 +88,52 @@ getAllDhallFiles baseDir = do
         Text.Megaparsec.errorBundlePretty err <>
         "... documentation won't be generated for this file"
 
+    extractTypeIfInSource :: Expr Void Import -> Maybe (Expr Void Import)
+    extractTypeIfInSource expr = do
+        V name index <- maybeNameInLet expr
+        (Binding _ _ _ (Just (_, exprType)) _ _) <-
+            getLetBindingWithIndex index $ getLetBindingsWithName name
+        return exprType
```

Comment:
I'm somewhat suspect that this could be simplified quite a bit now by using `Dhall.Syntax.multiLet`.

Reply:
Indeed